### PR TITLE
Add support for throwing errors with custom error code

### PR DESCRIPTION
**Pull request by palamccc**
### Motivation:

In the rsocket-js implementation (v0 branch), all stream errors are serialized and transmitted to other end [with fixed error code](https://github.com/rsocket/rsocket-js/blob/cdfa824513ae58c5ef8abb14c3754c99199296b8/packages/rsocket-core/src/RSocketMachine.js#L870). Due to this bug, there is no way to send custom error codes over stream (Bug report https://github.com/rsocket/rsocket-js/issues/218)

Custom error codes are supported in the rsocket protocol and implemented in rsocket-java this using [RSocketErrorException](https://github.com/rsocket/rsocket-java/blob/6d0738974fc3451508112dfdaeee2e174666e1cc/rsocket-core/src/main/java/io/rsocket/RSocketErrorException.java#L43) . When the application throws `RSocketErrorExecption`, the error code of the exception is sent over the stream. But this feature is not implemented in rsocket-js.

### Modifications:

- This PR adds custom Javascript Error class `RSocketError` in `RSocketMachine.js`.
- When the application throws error of type `RSocketError`, the` errorCode` of the error object is sent over the stream.

### Result:

This will allow application to send custom error codes to other party.

### CODE REVIEW

**Comment by viglucci** [comment]
Hey @palamccc thanks for the contribution!

Would you be able to add tests for your changes? Additionally the DCO check will ideally need to be resolved before this can be merged.

File: packages/rsocket-core/src/RSocketMachine.js [hunk 1]
```diff
@@ -943,3 +943,11 @@ function deserializeMetadataPushPayload<D, M>(
     metadata: serializers.metadata.deserialize(frame.metadata),
   };
 }
+
+export class RSocketError extends Error {
```
Comment (viglucci) [inline packages/rsocket-core/src/RSocketMachine.js:947]: This might be best as its own file `RSocketError.js`.

Comment (palamccc) [reply packages/rsocket-core/src/RSocketMachine.js]: Done 👍🏽

**Comment by palamccc** [comment]
> Would you be able to add tests for your changes? Looks like `RSocketServer-test.js` might have good examples to build off of.

@viglucci  I've added tests now.

**Comment by viglucci** [comment]
LGTM. Thanks!

**Comment by palamccc** [comment]
I've contributed types to DefinitelyTyped https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68788
Waiting for 0.0.28 release :-).

**Comment by viglucci** [comment]
@palamccc I've published a new version which should include this change. If you would be so kind, please validate this change with `rsocket-core@0.0.29-alpha.0`.

**Pull request merged**